### PR TITLE
[GEOT-5121] GeoTiff and WorldImage readers report incorrect geographic boundind box when reading overviews

### PR DESCRIPTION
**Pull request by aaime**

### CODE REVIEW

File: modules/library/coverage/src/main/java/org/geotools/coverage/grid/io/AbstractGridCoverage2DReader.java [hunk 1]
```diff
@@ -1233,5 +1241,45 @@ public GroundControlPoints getGroundControlPoints(String coverageName) {
...
+        //
+        // setting new coefficients to define a new affineTransformation
+        // to be applied to the grid to world transformation
+        // -----------------------------------------------------------------------------------
+        //
+        // With respect to the original envelope, the obtained planarImage
+        // needs to be rescaled. The scaling factors are computed as the
+        // ratio between the output raster sizes and the original sizes
+        // (this correctly accounts for
+        // //
```
Comment (dromagnoli) [inline modules/library/coverage/src/main/java/org/geotools/coverage/grid/io/AbstractGridCoverage2DReader.java]: for ... ? :)

File: modules/library/coverage/src/main/java/org/geotools/coverage/grid/io/AbstractGridCoverage2DReader.java [hunk 2]
```diff
@@ -1233,5 +1241,45 @@ public GroundControlPoints getGroundControlPoints(String coverageName) {
         }
         return null;
     }
+    
+    /**
+     * Computes the raster to model of a rescaled output raster, based on the original transform and
+     * output raster scaling facto
+     *
```
Comment (dromagnoli) [inline modules/library/coverage/src/main/java/org/geotools/coverage/grid/io/AbstractGridCoverage2DReader.java]: trivial TYPO: facto -> factor

**Comment by aaime** [comment]
Typos fixed, thanks for the review

**Comment by dromagnoli** [comment]
beside these small typos, the code looks good. +1 to merge


On Tue, May 19, 2015 at 11:58 AM, Andrea Aime <notifications@github.com>
wrote:

> Typos fixed, thanks for the review
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/geotools/geotools/pull/845#issuecomment-103422226>.
>



-- 
==
Meet us at the INSPIRE Conference in Lisbon 25-29 May 2015! Visit
http://goo.gl/WHKDXT for more information.
==

Ing. Daniele Romagnoli
Senior Software Engineer

GeoSolutions S.A.S.
Via Poggio alle Viti 1187
55054  Massarosa (LU)
Italy
phone: +39 0584 962313
fax:      +39 0584 1660272

http://www.geo-solutions.it
http://twitter.com/geosolutions_it

-------------------------------------------------------

*AVVERTENZE AI SENSI DEL D.Lgs. 196/2003*

Le informazioni contenute in questo messaggio di posta elettronica e/o
nel/i file/s allegato/i sono da considerarsi strettamente riservate. Il
loro utilizzo è consentito esclusivamente al destinatario del messaggio,
per le finalità indicate nel messaggio stesso. Qualora riceviate questo
messaggio senza esserne il destinatario, Vi preghiamo cortesemente di
darcene notizia via e-mail e di procedere alla distruzione del messaggio
stesso, cancellandolo dal Vostro sistema. Conservare il messaggio stesso,
divulgarlo anche in parte, distribuirlo ad altri soggetti, copiarlo, od
utilizzarlo per finalità diverse, costituisce comportamento contrario ai
principi dettati dal D.Lgs. 196/2003.



The information in this message and/or attachments, is intended solely for
the attention and use of the named addressee(s) and may be confidential or
proprietary in nature or covered by the provisions of privacy act
(Legislative Decree June, 30 2003, no.196 - Italy's New Data Protection
Code).Any use not in accord with its purpose, any disclosure, reproduction,
copying, distribution, or either dissemination, either whole or partial, is
strictly forbidden except previous formal approval of the named
addressee(s). If you are not the intended recipient, please contact
immediately the sender by telephone, fax or e-mail and delete the
information in this message that has been received in error. The sender
does not give any warranty or accept liability as the content, accuracy or
completeness of sent messages and accepts no responsibility  for changes
made after they were sent or for other risks which arise as a result of
e-mail transmission, viruses, etc.

**Pull request merged**